backend/sync/service.go

```diff
@@ -202,25 +188,41 @@ func (s *Service) startSourceSync(ctx context.Context, source SourceConfig, inde
 	slog.Info("Starting periodic sync for source", "source", sourceInfo, "interval", interval)
 
 	// Initial sync
-	if err := s.SyncSource(ctx, source); err != nil {
-		slog.Error("Initial sync failed", "source", sourceInfo, "error", err)
+	status := s.SyncSourceWithStatus(ctx, source)
+	if status.Status == StatusFailed {
+		slog.Error("Initial sync failed", "source", sourceInfo, "error", *status.LastError)
 	}
+	s.updateStatusFromSyncResult(index, status)
 
 	for {
 		select {
 		case <-ctx.Done():
 			slog.Info("Stopping sync for source", "source", sourceInfo)
 			return
 		case <-ticker.C:
-			if err := s.SyncSource(ctx, source); err != nil {
-				slog.Error("Sync failed", "source", sourceInfo, "error", err)
+			status := s.SyncSourceWithStatus(ctx, source)
+			if status.Status == StatusFailed {
+				slog.Error("Sync failed", "source", sourceInfo, "error", *status.LastError)
 			}
+			s.updateStatusFromSyncResult(index, status)
 		}
 	}
 }
 
 // SyncSource performs a full sync for a single source
-func (s *Service) SyncSource(ctx context.Context, source SourceConfig) error {
+// Returns the number of components discovered during sync
+func (s *Service) SyncSource(ctx context.Context, source SourceConfig) (int, error) {
```

Comment:
Let's only keep one method. Rename SyncSourceWithStatus to SyncSource and remove this one.